File: mrcomic-theme-platform/src/models/Comment.js
const mongoose = require('mongoose');

const commentSchema = new mongoose.Schema({
  themeId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Theme',
    required: true
  },
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  parentId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Comment',
    default: null
  },
  content: {
    type: String,
    required: true,
    trim: true,
    maxlength: 1000
  },
  likes: {
    type: Number,
    default: 0
  },
  likedBy: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User'
  }]
}, {
  timestamps: true
});

// Индексы для быстрого поиска
commentSchema.index({ themeId: 1, createdAt: -1 });
commentSchema.index({ parentId: 1 });

const Comment = mongoose.model('Comment', commentSchema);

module.exports = Comment;

File: mrcomic-theme-client/app/src/main/java/com/example/mrcomic/theme/data/model/Comment.kt
package com.example.mrcomic.theme.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName

/**
 * Модель данных для комментария к теме
 */
@Entity(tableName = "comments")
data class Comment(
    @PrimaryKey
    @SerializedName("id")
    val id: String,
    
    @SerializedName("themeId")
    val themeId: String,
    
    @SerializedName("userId")
    val userId: String,
    
    @SerializedName("parentId")
    val parentId: String?,
    
    @SerializedName("content")
    val content: String,
    
    @SerializedName("createdAt")
    val createdAt: String,
    
    @SerializedName("updatedAt")
    val updatedAt: String,
    
    @SerializedName("likes")
    val likes: Int,
    
    @SerializedName("likedBy")
    val likedBy: List<String>?,
    
    @SerializedName("user")
    val user: User?
)
